core/vision/src/main/java/com/arcrobotics/ftclib/vision/UGAngleHighGoalPipeline.java

```diff
@@ -0,0 +1,124 @@
+package com.arcrobotics.ftclib.vision;
+
+import org.opencv.core.Mat;
+
+public class UGAngleHighGoalPipeline extends UGBasicHighGoalPipeline {
+
+    class Fraction {
+        private int numerator, denominator;
+
+        Fraction(long a, long b) {
+            numerator = (int) (a / gcd(a, b));
+            denominator = (int) (b / gcd(a, b));
+        }
+
+        /**
+         * @return the greatest common denominator
+         */
+        private long gcd(long a, long b) {
+            return b == 0 ? a : gcd(b, a % b);
+        }
+
+        public int getNumerator() {
+            return numerator;
+        }
+
+        public int getDenominator() {
+            return denominator;
+        }
+    }
+
+
+    private boolean isFirstFrame;
+
+    // Camera Settings
+    private int imageWidth;
+    private int imageHeight;
+
+    private double cameraPitchOffset;
+    private double cameraYawOffset;
+
+    private double fov;
+    private double imageArea;
+    private double centerX;
+    private double centerY;
+    private double horizontalFocalLength;
+    private double verticalFocalLength;
+
+    public enum Target {
+        RED, Blue
+    }
+
+    public UGAngleHighGoalPipeline(double fov, double cameraPitchOffset, double cameraYawOffset) {
+        super();
+        this.fov = fov;
+        this.cameraPitchOffset = cameraPitchOffset;
+        this.cameraYawOffset = cameraYawOffset;
+    }
+
+    public UGAngleHighGoalPipeline(double fov) {
+        this(fov, 0, 0);
+    }
+
+
+    @Override
+    public Mat processFrame(Mat input) {
+
+        if (isFirstFrame) {
```

Comment:
Does the entire pipeline in `UGHighAngleGoal` even run? The entire `processFrame` function is wrapped in an `isFirstFrame` boolean whose value is never set and never changed.

Reply:
Also y'all got some random new lines everyyyywhere

Reply:
Hmm must've missed the `isFirstFrame` in my initial review. Yeah, it's only ever called twice -- at the declaration and the if statement. Looks like it will just always call the `processFrame` of the `UGBasicHighGoalPipeline`.
Also, yeah, format code using the AS shortcut.

Reply:
Formatting code doesn't remove extraneous newlines

Reply:
Not by default, but you can change it in settings iirc.

Reply:
You can. I just use google's settings unmodified tho 🤷 
Enabling the blank line removal is a good idea tho.